src/components/MessagesList/MessagesList.vue

```diff
@@ -318,6 +322,18 @@ export default {
 	},
 
 	methods: {
+		softUpdateMessagesList(oldList, newList) {
+			const oldGroupsMap = new Map(oldList.map(group => [group.id, { ...group }]))
+
+			// Check if we have this group in the old list already and it is unchanged
+			return newList.map(newGroup => oldGroupsMap.has(newGroup.id)
+				&& newGroup.messages.length === oldGroupsMap.get(newGroup.id).messages.length
+				&& newGroup.dateSeparator === oldGroupsMap.get(newGroup.id).dateSeparator
+				? { ...oldGroupsMap.get(newGroup.id) }
+				: { ...newGroup }
+			).sort((a, b) => a.id - b.id)
+		},
```

Comment:
> By this change we can assure, that messagesGroupedByAuthor receives the same object to form a group as before

We don't. New groups recreated by `{ ...newGroup }`, old group without changes recreated twice, first by `{ ...group }` and then by `{ ...oldGroupsMap.get(newGroup.id) }`.

Only objects inside `messagesGroupedByAuthor[].messages[]` are unchanged.

The question is, do we need object cloning here on each step?

Reply:
`newList` from this method will be created one way or another, because we have to check, if the newly fetched messages belong to one of existing groups.

> The question is, do we need object cloning here on each step?

I didn't test it yet, but I think you've meant to change these lines? Should work, then it's my fault to overlook it:

`oldList.map(group => [group.id, { ...group }])` => `oldList.map(group => [group.id, group])`
`? { ...oldGroupsMap.get(newGroup.id) }` => `? oldGroupsMap.get(newGroup.id)`
`: { ...newGroup }` => `: newGroup`

Reply:
Replaced with described code, seems to still work, and shouldn't create copies of object, so let's give it a try